omnibus/files/private-chef-cookbooks/private-chef/providers/pg_upgrade.rb

```diff
@@ -123,6 +124,37 @@ def upgrade_required?
   end
 end
 
+#
+# Since we don't use the --link flag, we need to ensure the disk has
+# enough space for another copy of the postgresql data.
+#
+def check_required_disk_space
+  old_data_dir_size = Du.du(old_data_dir)
+  # new_data_dir might not exist at the point of making this check.
+  # In that case check the first existing directory above it.
+  new_dir = dir_or_existing_parent(new_data_dir)
+  free_disk_space = Statfs.new(new_dir).free_space
+
+  if old_data_dir_size < (free_disk_space * 0.90)
+    Chef::Log.debug("Old data dir size: #{old_data_dir_size}")
+    Chef::Log.debug("  Free disk space: #{free_disk_space}")
+    Chef::Log.debug("Free space is sufficient to start upgrade")
+    true
+  else
+    Chef::Log.error("Insufficient free space on disk to complete upgrade.")
+    Chef::Log.error("The current postgresql data directory contains #{old_data_dir_size} KB of data but only #{free_disk_space} KB is available on disk.")
+    Chef::Log.error("The upgrade process requires at least #{free_disk_space/0.90} KB.")
```

Comment:
I think I've just goofed and this should have been `old_data_dir_size/0.9`. That is, I used the wrong variable.